### PR TITLE
New API Endpoints

### PR DESCRIPTION
**Pull request by Raj-Shah1**
### New Features and Enhancements:

- API - Edit Note Endpoint [#18](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/18)
- API - Get a list of Events in an account [#29](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/29)
- API - Create Event for an Account [#30](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/30)
- API - Edit Event in an Account Endpoint [#31](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/31)
- API - Delete Event in an Account Endpoint [#32](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/32)
- API - Edit Task in an Account Endpoint [#36](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/36)
- API - Get Event By Id Endpoint [#101](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/101)
- API - Get Task By Id Endpoint [#102](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/102)
- Add Event Suggestion Prompt [#103](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/103)

@coderabbitai: ignore

### CODE REVIEW

File: src/main/java/com/salessparrow/api/lib/openAi/OpenAiPayloadBuilder.java [hunk 1]
```diff
@@ -17,29 +17,43 @@ public class OpenAiPayloadBuilder {
...
+				+ "      \"content\": \"You are an AI assistant that provides suggestions for creating tasks and events in CRM based solely on the content of the input message. The content of the task or event, if any found, should only be from the input message. If no task or event suggestions are found in the input message, return empty data. Suggestions can either be both task and event list or only tasks or only events or empty. If task suggestions are found, they should include description and due date. Due Date format should be YYYY-MM-DD. If event suggestions are found, they should include description and start datetime and end datetime. If end datetime not provided it should be start datetime + 1 hour. Start datetime and end datetime format is yyyy-MM-dd'T'HH:mm:ss.SSS+0000. Today's date is "
```
Comment (AMAN-BARBARIA) [inline src/main/java/com/salessparrow/api/lib/openAi/OpenAiPayloadBuilder.java:25]: If end datetime not provided it should be start datetime + 1 hour.
Should this be done programatically? What is the success rate of GPT for such cases?

File: src/test/resources/data/functional/controllers/accountEventController/updateAccountEvent.scenarios.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+      "description": "Should successfully update the event for given event id", 
+      "input": {
+        "body": {
+          "description":"Create Event",
+          "start_datetime": "2023-07-21T13:12:17.000+0000",
+          "end_datetime": "2023-07-22T13:12:17.000+0000"
+        },
+        "accountId": "0011e00000bWSxdAAG",
+        "eventId":"0691e000001X1yTAAS"
```
Comment (AMAN-BARBARIA) [inline src/test/resources/data/functional/controllers/accountEventController/updateAccountEvent.scenarios.json:11]: Should there be test cases for start_datetime & end_datetime?

File: src/test/resources/data/functional/controllers/accountTaskController/updateAccountTask.scenarios.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    {
+      "description": "Should successfully update the task for given task id", 
+      "input": {
+        "body": {
+          "crm_organization_user_id":"0011e00000bWSxdAA1",
+          "description":"Update remaining documentations",
+          "due_date":"2023-12-01"
+        },
+        "accountId": "0011e00000bWSxdAAG",
+        "taskId":"0691e000001X1yTAAS"
```
Comment (AMAN-BARBARIA) [inline src/test/resources/data/functional/controllers/accountTaskController/updateAccountTask.scenarios.json:11]: Add Test Cases for description/due date/crm_organization_user_id

**Review by AMAN-BARBARIA — COMMENTED** [review]
Review comments added

File: src/main/java/com/salessparrow/api/lib/crmActions/createAccountNote/CreateSalesforceNote.java [hunk 4]
```diff
@@ -51,13 +51,15 @@ public class CreateSalesforceNote implements CreateNoteInterface {
...
 	 */
-	public CreateNoteFormatterDto createNote(SalesforceUser user, String accountId, NoteDto note) {
+	public CreateNoteFormatterDto createNote(SalesforceUser user, String accountId, AccountNoteDto note) {
 		String salesforceUserId = user.getExternalUserId();
 
-		Util util = new Util();
 		String noteContent = note.getText();
-		String noteTitle = getNoteTitleFromContent(noteContent);
-		noteContent = util.replaceNewLineWithBreak(noteContent);
+		String unEscapeNoteContent = Util.unEscapeSpecialCharactersForPlainText(noteContent);
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/main/java/com/salessparrow/api/lib/crmActions/createAccountNote/CreateSalesforceNote.java:58]: It was already there in getNoteTitleFromContent method, I just moved that logic here and changed into two parts unEscapeSpecialCharactersForPlainText and getTrimmedString.

File: src/main/java/com/salessparrow/api/lib/openAi/OpenAiPayloadBuilder.java (hunk 1 above)
Comment (Raj-Shah1) [reply src/main/java/com/salessparrow/api/lib/openAi/OpenAiPayloadBuilder.java:25]: Till now, it is giving almost 100% success rate, And yes we can move that to be done programatically, that will also save some token data, will add it as changes for new spring

File: src/test/resources/data/functional/controllers/accountEventController/updateAccountEvent.scenarios.json (hunk 2 above)
Comment (Raj-Shah1) [reply src/test/resources/data/functional/controllers/accountEventController/updateAccountEvent.scenarios.json]: Added

File: src/test/resources/data/functional/controllers/accountTaskController/updateAccountTask.scenarios.json (hunk 3 above)
Comment (Raj-Shah1) [reply src/test/resources/data/functional/controllers/accountTaskController/updateAccountTask.scenarios.json:11]: Added

**Review by AMAN-BARBARIA — APPROVED** [review]
LGTM

**Pull request merged**